ignite/metrics/gan/inception_score.py

```diff
@@ -0,0 +1,123 @@
+            you want to compute the metric with respect to one of the outputs.
+        device: specifies which device updates are accumulated on. Setting the metric's
+            device to be the same as your ``update`` arguments ensures the ``update`` method is non-blocking. By
+            default, CPU.
+
+    Example:
+
+    .. code-block:: python
+
+        from ignite.metrics.gan import InceptionScore
+
+        m = InceptionScore(splits=4)
+
+        y_pred = torch.rand((16,3,299,299))  # size must match the input shape of the InceptionV3 model
+
+        m.update(y_pred)
+
+        print(m.compute())
+
+    """
+
+    def __init__(
+        self,
+        splits: int = 10,
+        inception_model: Optional[nn.Module] = None,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+    ):
+        if inception_model is None:
+            try:
+                from torchvision import models  # type: ignore
+
+                inception_model = models.inception_v3(pretrained=True, transform_input=False)
+            except ImportError:
+                raise RuntimeError(
+                    "This metric requires torchvision to be installed. "
+                    "You may install torchvision using: \n pip install torchvision \n"
+                )
+        if not isinstance(inception_model, nn.Module):
+            raise TypeError(
+                f"Argument inception_model should be instance of nn.Module, but given {type(inception_model)}"
+            )
+        super(InceptionScore, self).__init__(output_transform=output_transform, device=device)
+        self.inception_model = inception_model.eval().to(self._device)
+        self.n_splits = splits
+
+    @reinit__is_reduced
+    def reset(self) -> None:
+        self._probs = []  # type: List[torch.Tensor]
+
+    @reinit__is_reduced
+    def update(self, output: torch.Tensor) -> None:
+        generated = output.detach()
+        inception_output = self.inception_model(generated)
+        probs = F.softmax(inception_output)
+        self._probs.append(probs)
+
+    def compute(self) -> float:
+        if len(self._probs) < 1:
+            raise NotComputableError("Inception score must have at least one example before it can be computed.")
+
+        ws = idist.get_world_size()
+        _probs_tensor = torch.cat(self._probs, dim=0)
+
+        if ws > 1 and not self._is_reduced:
+            _probs_tensor = cast(torch.Tensor, idist.all_gather(_probs_tensor))
+        self._is_reduced = True
+
+        result = 0.0
+        if idist.get_rank() == 0:
+            N = _probs_tensor.shape[0]
+            scores = torch.zeros((self.n_splits,))
+            for i in range(self.n_splits):
+                part = _probs_tensor[i * (N // self.n_splits) : (i + 1) * (N // self.n_splits)]
+                kl = part * (torch.log(part) - torch.log(torch.mean(part, dim=0)))
```

Comment:
Please, if we could reuse pytorch functions to compute KL. We can also probably avoid applying softmax in the `update` pass.